tests/core/data_layer/test_data_store.py

```diff
@@ -1119,17 +1097,72 @@ async def test_rollback_to_generation(data_store: DataStore, tree_id: bytes32) -
 
 @pytest.mark.asyncio
 async def test_subscribe_unsubscribe(data_store: DataStore, tree_id: bytes32) -> None:
-    await data_store.subscribe(Subscription(tree_id, DownloadMode.HISTORY, "127.0.0.1", uint16(8000)))
-    assert await data_store.get_subscriptions() == [
-        Subscription(tree_id, DownloadMode.HISTORY, "127.0.0.1", uint16(8000))
-    ]
-    await data_store.update_existing_subscription(Subscription(tree_id, DownloadMode.HISTORY, "0.0.0.0", uint16(8000)))
-    assert await data_store.get_subscriptions() == [
-        Subscription(tree_id, DownloadMode.HISTORY, "0.0.0.0", uint16(8000))
-    ]
-    await data_store.update_existing_subscription(Subscription(tree_id, DownloadMode.HISTORY, "0.0.0.0", uint16(8001)))
-    assert await data_store.get_subscriptions() == [
-        Subscription(tree_id, DownloadMode.HISTORY, "0.0.0.0", uint16(8001))
-    ]
+    await data_store.subscribe(Subscription(tree_id, ["http://127:0:0:1/8000"]))
+    assert await data_store.get_subscriptions() == [Subscription(tree_id, ["http://127:0:0:1/8000"])]
+    await data_store.update_existing_subscription(Subscription(tree_id, ["http://0.0.0.0/8002"]))
+    assert await data_store.get_subscriptions() == [Subscription(tree_id, ["http://0.0.0.0/8002"])]
+    await data_store.update_existing_subscription(Subscription(tree_id, ["https://0.0.0.0/8001"]))
+    assert await data_store.get_subscriptions() == [Subscription(tree_id, ["https://0.0.0.0/8001"])]
     await data_store.unsubscribe(tree_id)
     assert await data_store.get_subscriptions() == []
+    await data_store.subscribe(Subscription(tree_id, ["http://127:0:0:1/8003", "http://127:0:0:1/8004"]))
+    assert await data_store.get_subscriptions() == [
+        Subscription(tree_id, ["http://127:0:0:1/8003", "http://127:0:0:1/8004"])
+    ]
+
+
+@pytest.mark.parametrize(
+    "test_delta",
+    [True, False],
+)
+@pytest.mark.asyncio
+async def test_data_server_files(data_store: DataStore, tree_id: bytes32, test_delta: bool, tmpdir: Path) -> None:
+    roots: List[Root] = []
+    num_batches = 10
+    num_ops_per_batch = 100
+
+    temp_directory_path = Path(tmpdir)
```

Comment:
The `tmp_path` fixture is already a `Path`, if I remember correctly.  Likewise in the test above.